External/Plugins/ASCompletion/Completion/ASComplete.cs

```diff
@@ -4439,6 +4439,58 @@ private static bool IsXmlType(ClassModel model)
                 && (model.QualifiedName == "XML" || model.QualifiedName == "XMLList");
         }
 
+        public static int ExpressionEndPosition(ScintillaControl sci, int position)
+        {
+            var member = ASContext.Context.CurrentMember;
+            var endPosition = member != null ? sci.PositionFromLine(member.LineTo) : sci.TextLength;
+            return ExpressionEndPosition(sci, position, endPosition);
+        }
+
+        public static int ExpressionEndPosition(ScintillaControl sci, int startPos, int endPos)
+        {
+            var characterClass = ScintillaControl.Configuration.GetLanguage(sci.ConfigurationLanguage).characterclass.Characters;
+            var groupCount = 0;
+            var spacesCount = 0;
+            var statementEnd = startPos;
+            var hadWS = false;
+            sci.Colourise(0, -1);
+            while (statementEnd < endPos)
+            {
+                if (sci.PositionIsOnComment(statementEnd) || sci.PositionIsInString(statementEnd))
+                {
+                    if (groupCount == 0) spacesCount++;
+                    statementEnd++;
+                    continue;
+                }
+                var c = (char) sci.CharAt(statementEnd++);
+                if (c == '(')
+                {
+                    groupCount++;
+                    hadWS = false;
+                    spacesCount = 0;
+                }
+                else if (c == ')')
+                {
+                    groupCount--;
+                    if (groupCount < 0) break;
+                }
+                else if (groupCount == 0)
+                {
+                    if (characterClass.Contains(c))
+                    {
+                        if (hadWS) break;
+                    }
+                    else if (c <= ' ')
+                    {
+                        hadWS = true;
+                        spacesCount++;
+                    }
+                    else break;
+                }
+            }
+            return statementEnd - 1 - spacesCount;
```

Comment:
Why substracting the spaces? Don't they still count in the character position of the expression?
Shouldn't you return the word-end position in case `groupCount` doesn't reach `0`?

Reply:
`a += b`
groupCount is always zero